From 11ef8cf5a5526cfd855981519388265a20e0ecaf Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Thu, 2 May 2013 16:55:00 +0800
Subject: libstagefright: avoid crash by checking the configuration vserion

BZ: 94702

The value of configuration version in avcc box should be 1 but some
malformed clip contains other value and causes mediaserver crash.

Category: aosp improvement
Domain: Video.Media-container
Origin: internal
Upstream-Candidate: yes

Change-Id: I12a301d31b88dc9643075183fcb22b7ae9171bbe
Orig-Change-Id: Id820b13534c161b8d95580d864e084ea131f7a27
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 17a8d24..db25516 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -1553,6 +1553,10 @@ status_t MPEG4Extractor::parseChunk(off64_t *offset, int depth) {
                 return ERROR_IO;
             }
 
+            // configurationVersion should be 1
+            if (buffer->data()[0] != 1)
+                return ERROR_MALFORMED;
+
             mLastTrack->meta->setData(
                     kKeyAVCC, kTypeAVCC, buffer->data(), chunk_data_size);
 
-- 
1.7.4.1

